packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -966,7 +966,14 @@ module.exports = (
       path: '/account/profile',
       options: {
         auth: {
-          strategies: ['sessionToken', 'oauthToken'],
+          strategies: ['sessionToken', 'oauthToken', 'profileSecret'],
+        },
+        validate: {
+          query: {
+            client_id: isA.string().optional(),
+            scope: isA.string().optional(),
+            uid: isA.string().optional(),
```

Comment:
If you didn't want to put these in the query params, you could bundle them up into a JWT and sign that with the shared secret, then have the `profileSecret` auth strategy extract and validate them for you. We already have the pieces needed to do that, because it's basically how auth-server talks to oauth-server right now.

Reply:
I went down that route, but stopped because the auth server passes the JWT in [POST body params](https://github.com/mozilla/fxa/blob/a142c2528ea83dfd2852e0e5fa9cd79e74193904/packages/fxa-auth-server/lib/oauthdb/index.js#L117) and since this is a GET request, we'd still have to pass the JWT as a query param. That, or I'd have to modify backendService.js to accept a header as part of the request instead of when [creating the Service](https://github.com/mozilla/fxa/blob/a142c2528ea83dfd2852e0e5fa9cd79e74193904/packages/fxa-auth-server/lib/backendService.js#L86). Seemed like a bunch of extra hoops to go through to clean up code we want to soon tear out.

Reply:
So, I was just exploring with modifying backendService.js and that's not as bad as I expected, but then realized the way I implemented jwt creation and validation makes assumptions about which key to use, and then over on the auth server, we'd have to use the oauth server's jwt validation logic. It's a bit of a mess really.